composables/usePrefix.ts

```diff
@@ -10,57 +10,35 @@ const sharedPrefix = ref<Prefix>()
 export default function () {
   const route = useRoute()
   const storage = useLocalStorage('urlPrefix', { selected: DEFAULT_PREFIX })
-  const initialPrefixFromPath = getAvailablePrefix(route.path.split('/')[1])
-  const validPrefixFromRoute = computed(() =>
-    getAvailablePrefix(route.params.prefix)
-  )
 
-  const prefix = computed<Prefix>(
+  const urlPrefix = computed<Prefix>(
     () =>
       (sharedPrefix.value ||
-        validPrefixFromRoute.value ||
+        getAvailablePrefix(route.params.prefix) ||
         storage.value.selected ||
-        initialPrefixFromPath) as Prefix
-  )
-
-  watch(
-    prefix,
-    (value) => {
-      if (value) {
-        sharedPrefix.value = value
-        storage.value = { selected: value }
-      }
-    },
-    { immediate: true }
+        getAvailablePrefix(route.path.split('/')[1])) as Prefix
   )
```

Comment:
What happens if user changes url ? 
will be the old prefix stuck in the local storage?

Reply:
If the user changes route, it is mostly done by setUrlPrefix all throughout the codebase. So yes, it will persist.

Reply:
That's tbh not very good.

Reply:
I agree @vikiival, 
Would you want me to refactor all the these as well and handle setting of prefix when you use navigateTo?
So we don't need to call setUrlPrefix anymore when navigating?